server/src/test/java/org/opensearch/action/bulk/TransportShardBulkActionTests.java

```diff
@@ -947,6 +948,68 @@ public void testRetries() throws Exception {
         latch.await();
     }
 
+    public void testUpdateWithRetryOnConflict() throws IOException, InterruptedException {
+        IndexSettings indexSettings = new IndexSettings(indexMetadata(), Settings.EMPTY);
+        UpdateRequest writeRequest = new UpdateRequest("index", "id").doc(Requests.INDEX_CONTENT_TYPE, "field", "value");
+        writeRequest.retryOnConflict(3);
+        BulkItemRequest primaryRequest = new BulkItemRequest(0, writeRequest);
```

Comment:
Also, can we have couple items without the retry in the end for sanity. Also, do you think, it would be better to simulate two separate bulk requests? The second bulk requests may or may not have retries configured (to verify the reset as mgodwan mentioned)

Reply:
> Also, can we have couple items without the retry in the end for sanity.

Yes @sarthakaggarwal97 that should be useful.

> Also, do you think, it would be better to simulate two separate bulk requests? The second bulk requests may or may not have retries configured (to verify the reset as mgodwan mentioned)

Simulating 2 bulk shard requests should not be a problem, but it might not have much of an impact since the BulkPrimaryExecutionContext is unique to each individual execution. Do you think there is any other benefit?

> Can we add multiple items here and verify that the retry is reset and works for each expected item?

Yes @mgodwan I'm trying to see how we can assert the retry count for each item.

Reply:
I have added an assertion for how many times a request is retried and I have added multiple items in a single shard bulk request.